### ISSUE TITLE
KEDA using Azure Service Bus - scaling down criteria

### ISSUE THREAD

**Issue by ronlhayes** [microsoft/azure-container-apps#557]
This is more of a question than an issue. I have a .Net 6 service that uses Azure Service Bus (via MassTransit) that I would like to port to an Azure Container App. The consumer can take anywhere from 20-30 seconds up to 30 minutes to process a single message. Setting up KEDA to scale up using a message count of 1 should create a new replica for each message up to 30. My concern is how KEDA decides when to scale down a replica. What criteria is used? CPU usage? A fixed timeout? Thanks in advance for the help.

**Comment by tomkerkhove** [comment]
KEDA will start scaling in (not down) when the queue is empty and start killing instances.

It looks like you are doing "long-running" processing for which ScaledJobs is built; which is tracked in #526

**Comment by ronlhayes** [comment]
Yes, that sounds like a better choice. For future reference, when using the Azure Service Bus (ASB) scaler is there an assumed timeout for each replica when scaling in (killing instances). For example 20 messages arrive in a ASB queue and the scaler is set to create a new replica for every 1 message. Would 20 replicas get created? Assuming each takes 5-10 seconds to complete, once the queue count reaches 0, does the scaler use the start time of each replica and have a defined cool down before killing? If so, what is that cool down/timeout?

Regarding the ScaledJob scaler. Assuming I use a ASB trigger and have the same scenario as above. Would the scaler create 20 replicas/jobs? Each replica/job would read a single ASB message, complete the associated work, remove the ASB message,  and then exit? Once the message count reaches the scaler would scale in using the pollingInterval and remove replicas based on the successful/failed job limits? How does a job indicate success/failure when exiting?

**Comment by Black0-shark** [comment]
+1, we are having the similar issue.
